federation/cluster/log-dump.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Call this to dump all Federation pod logs into the folder specified in $1
+# (defaults to _artifacts).
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# For FEDERATION_NAMESPACE
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/federation/cluster/common.sh"
+
+readonly REPORT_DIR="${1:-_artifacts}"
+OUTPUT_DIR="${REPORT_DIR}/federation"
+
+# Dumps logs for all pods in a federation.
+function dump_federation_pod_logs() {
+  local -r federation_pod_names=($(kubectl get pods -l 'app=federated-cluster' --namespace=${FEDERATION_NAMESPACE} -o name))
+  for pod_name in ${federation_pod_names[@]}; do
```

Comment:
I see the following in logs:
```
./federation/cluster/log-dump.sh: line 34: federation_pod_names[@]: unbound variable
```
Example run: http://prow.k8s.io/log?pod=pull-kubernetes-federation-e2e-gce-1583